spec/models/contact/merge_spec.rb

```diff
@@ -0,0 +1,170 @@
+    context 'update labels' do
+      it do
+        base_contact.update(label_list: "label1, label2")
+        mergee_contact.update(label_list: "label2, label3")
+
+        merge_service.send(:merge_labels)
+        expect(base_contact.label_list).to match_array(["label1", "label2", "label3"])
+      end
+      it do
+        base_contact.update(label_list: "label1, label2")
+
+        merge_service.send(:merge_labels)
+        expect(base_contact.label_list).to match_array(["label1", "label2"])
+      end
+      it do
+        mergee_contact.update(label_list: "label2, label3")
+
+        merge_service.send(:merge_labels)
+        expect(base_contact.label_list).to match_array(["label2", "label3"])
+      end
+    end
+    it 'update chatwoot_conversations_labels' do
+      base_contact.update(chatwoot_conversations_label_list: "label_conversation_1, label_conversation_2")
+      mergee_contact.update(chatwoot_conversations_label_list: "label_conversation_2, label_conversation_3" )
+
+      merge_service.send(:merge_labels)
+      expect(base_contact.chatwoot_conversations_label_list).to match_array(["label_conversation_1", "label_conversation_2", "label_conversation_3"])
+    end
+    it 'should not update labels' do
+      merge_service.send(:merge_labels)
+      expect(base_contact.label_list).to match_array([])
+      expect(base_contact.chatwoot_conversations_label_list).to match_array([])
+    end
+  end
+
+  describe '#merge_events' do
+    let!(:event) { create(:event, contact: mergee_contact) }
+    let!(:event2) { create(:event, contact: mergee_contact) }
+    let!(:event3) { create(:event, contact: base_contact) }
+
+    it 'updates event sender to base_contact' do
+      merge_service.send(:merge_events)
+      expect(event.reload.contact_id).to eq(base_contact.id)
+      expect(event2.reload.contact_id).to eq(base_contact.id)
+      expect(event3.reload.contact_id).to eq(base_contact.id)
+    end
+  end
+
+  describe '#merge_and_remove_mergee_contact' do
+    before do
+      base_contact.update(full_name: 'Base Name', additional_attributes: { field1: 'value1' })
+      mergee_contact.update(full_name: 'Mergee Name', additional_attributes: { field2: 'value2' })
+    end
+
+    it 'destroys mergee_contact' do
+      expect { merge_service.send(:merge_and_remove_mergee_contact) }
+        .to change { Contact.exists?(mergee_contact.id) }.from(true).to(false)
+    end
+
+    it 'updates base_contact with merged attributes' do
+      merge_service.send(:merge_and_remove_mergee_contact)
+      base_contact.reload
+      expect(base_contact.additional_attributes).to eq('field1' => 'value1', 'field2' => 'value2')
+    end
+
+    it 'prioritizes base_contact attributes' do
+      base_contact.update(email: 'base@example.com')
+      mergee_contact.update(email: 'mergee@example.com')
+
+      merge_service.send(:merge_and_remove_mergee_contact)
+      base_contact.reload
+      expect(base_contact.email).to eq('base@example.com')
+    end
```

Comment:
Não acho que é muito interessante fazer o update dentro do it dessa forma pois o model é criado e depois atualizado em vez de já ser criado no formato que vai ser utilizado, outro ponto ruim que eu vejo é que para eu entender o `base_contact` completamente preciso subir lá na linha 5

Segue uma sugestão:

```suggestion
    context 'prioritizes base_contact attributes' do
      let(:base_contact) { create(:contact, email: 'base@example.com') }
      let(:mergee_contact) { create(:contact, email: 'mergee@example.com') }

      it do
        merge_service.send(:merge_and_remove_mergee_contact)
        base_contact.reload
        expect(base_contact.email).to eq('base@example.com')
      end
    end
```